src/cache/webdav.rs

```diff
@@ -32,11 +34,19 @@ impl WebdavCache {
             .root(key_prefix)
             .username(username.unwrap_or_default())
             .password(password.unwrap_or_default())
-            .token(token.unwrap_or_default());
+            .token(token.unwrap_or_default())
+            .http_client(set_user_agent());
 
         let op = Operator::new(builder)?
             .layer(LoggingLayer::default())
             .finish();
         Ok(op)
     }
 }
+
+/// Set the user agent (helps with monitoring on the server side)
+fn set_user_agent() -> HttpClient {
```

Comment:
I recall that we already have similar code that builds an HTTP client. Would you like to check the codebase again?

Reply:
You're right! I've removed the copied code and referenced `set_user_agent` from `src/cache/s3.rs`.